cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -335,6 +335,12 @@ private Element adjustedSink(DataFlow::Node sink) {
   // For compatibility, send flow into a `NotExpr` even if it's part of a
   // short-circuiting condition and thus might get skipped.
   result.(NotExpr).getOperand() = sink.asExpr()
+  or
+  // Taint `e--` and `e++` when `e` is tainted.
+  result.(PostfixCrementOperation).getAnOperand() = sink.asExpr()
```

Comment:
I've been investigating the lost result (in defaulttainttracking) in the CWE-190 IntegerOverflowTainted.expected test in the internal semmlecode-cpp-tests.  I tracked it down to an issue with `adjustedSink` and it can be fixed by adding a case like this only for `PrefixCrementOperation` - or for both with `CrementOperation`.  Your explanation sounds similar to what I'm seeing as well - we're not reporting a a tainted variable being decremented in void context.

Are you happy that your fix could extend to the general `CrementOperation`?  If so, do you want to do it here or would you prefer if I made a follow-up PR?

Reply:
Interesting. I can do it here while I fix @jbj's comment https://github.com/Semmle/ql/pull/2799#discussion_r377110781. I'm surprised `PrefixCrementOperation` is not being propagated by `adjustedSink ` through `sink.asExpr().getConversion*() = result`, though.

Reply:
This PR seems like a good place to generalise to `CrementOperation`. The result of expression `++x` should be modelled the same way as `x += 1` (for all practical purposes), both in the AST and IR, and special support for `x += 1` was deemed necessary.